### PR TITLE
GEOMESA-2049 Orc file support in FSDS

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Refactored file-system code to allow pluggable FS implementations

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: build/cqs.tsv [hunk 1]
```diff
@@ -21,6 +21,7 @@ com.google.guava:guava	12.0.1	compile
 com.google.guava:guava	14.0.1	compile
 com.google.guava:guava	17.0	compile
 com.google.guava:guava	18.0	compile
+com.google.protobuf:protobuf-java	2.5.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: approved ~https://dev.eclipse.org/ipzilla/show_bug.cgi?id=10590~

File: build/cqs.tsv [hunk 2]
```diff
@@ -139,9 +142,12 @@ org.apache.hbase:hbase-client	1.3.1	compile
 org.apache.hbase:hbase-common	1.3.1	compile
 org.apache.hbase:hbase-protocol	1.3.1	compile
 org.apache.hbase:hbase-server	1.3.1	compile
+org.apache.hive:hive-storage-api	2.2.1	compile
 org.apache.htrace:htrace-core	3.1.0-incubating	compile
 org.apache.metamodel:MetaModel-core	4.3.6	compile
 org.apache.metamodel:MetaModel-pojo	4.3.6	compile
+org.apache.orc:orc-core	1.4.1	compile
+org.apache.orc:orc-mapreduce	1.4.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15255

File: build/cqs.tsv [hunk 3]
```diff
@@ -139,9 +142,12 @@ org.apache.hbase:hbase-client	1.3.1	compile
 org.apache.hbase:hbase-common	1.3.1	compile
 org.apache.hbase:hbase-protocol	1.3.1	compile
 org.apache.hbase:hbase-server	1.3.1	compile
+org.apache.hive:hive-storage-api	2.2.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15256

File: build/cqs.tsv [hunk 4]
```diff
@@ -65,6 +66,8 @@ de.topobyte:osm4j-tbo	0.0.9	compile
 de.topobyte:osm4j-utils	0.0.22	compile
 de.topobyte:osm4j-xml	0.0.3	compile
 eu.medsea.mimeutil:mime-util	2.1.3	compile
+io.airlift:aircompressor	0.8	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15257

File: build/cqs.tsv [hunk 5]
```diff
@@ -65,6 +66,8 @@ de.topobyte:osm4j-tbo	0.0.9	compile
...
+io.airlift:slice	0.29	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15258

**Comment by elahrvivaz** [comment]
there will be more conflicts once #1796 is merged, so I'll fix them then

**Comment by jnh5y** [comment]
@elahrvivaz Two quick requests (feel free to push back if it'd be too hard):  Can this be rebased now that the other PRs are in?  And could it be split up so that just the ORC work is in one top PR?

**Comment by jnh5y** [comment]
(I'm suggesting that so that we can review/merge as much of the background work while we waiting on CQs...)

**Comment by elahrvivaz** [comment]
@jnh5y yeah I had already pulled out the two PRs as background work. I just rebased now that those are merged, so this is just the ORC work now. All CQs have been approved, so we should be good there.

**Comment by elahrvivaz** [comment]
rebased

**Pull request merged**